modal/image.py

```diff
@@ -333,6 +339,13 @@ def pip_install_from_pyproject(
         pyproject_toml = os.path.expanduser(pyproject_toml)
 
         config = toml.load(pyproject_toml)
+        if optional_dependencies:
+            optionals = config["project"]["optional-dependencies"]
+            for dep in optional_dependencies:
+                if dep in optionals:
+                    config["project"]["dependencies"].extend(
```

Comment:
nit: although we're not using the config anywhere else, still slightly prefer making a separate list instead of mutating config

Reply:
👍